applications/dashboard/controllers/api/RolesApiController.php

```diff
@@ -9,21 +9,62 @@
 use Garden\Web\Exception\NotFoundException;
 use Garden\Web\Exception\ServerException;
 use Vanilla\Utility\CapitalCaseScheme;
+use Vanilla\Utility\CamelCaseScheme;
 
 /**
  * API Controller for the `/roles` resource.
  */
 class RolesApiController extends AbstractApiController {
 
+    /** @var CamelCaseScheme */
+    private $camelCaseScheme;
+
     /** @var CapitalCaseScheme */
     private $caseScheme;
 
+    /** @var array Groups of permissions that can be consolidated into one. */
+    private $consolidatedPermissions = [
+        'discussions.moderate' => ['discussions.announce', 'discussions.close', 'discussions.sink'],
+        'discussions.manage' => ['discussions.delete', 'discussions.edit']
+    ];
+
+    /** @var array Permissions that have been deprecated and should no longer be used. */
+    private $deprecatedPermissions = [
+        'Garden.Activity.Delete',
+        'Garden.Activity.View',
+        'Garden.SignIn.Allow',
+        'Garden.Curation.Manage',
+        'Vanilla.Approval.Require',
+        'Vanilla.Comments.Me'
+    ];
+
     /** @var Schema */
     private $idParamSchema;
 
     /** @var PermissionModel */
     private $permissionModel;
 
+    /** @var array A static mapping of updated permission names. */
+    private $renamedPermissions = [
+        'Conversations.Moderation.Manage' => 'conversations.moderate',
+        'Email.Comments.Add' => 'comments.email',
+        'Email.Conversations.Add' => 'conversations.email',
+        'Email.Discussions.Add' => 'discussions.email',
+        'Garden.Moderation.Manage' => 'community.moderate',
+        'Garden.NoAds.Allow' => 'noAds.use',
+        'Garden.Settings.Manage' => 'site.manage',
+        'Garden.Users.Approve' => 'applicants.manage',
+        'Groups.Group.Add' => 'groups.add',
+        'Groups.Moderation.Manage' => 'groups.moderate',
+        'Reputation.Badges.Give' => 'badges.moderate',
+        'Vanilla.Tagging.Add' => 'tags.add'
+    ];
+
+    /** @var array Permissions with a fixed name. Do not change them. */
+    private $fixedPermissions = [
```

Comment:
I don't understand why we have this array of permissions.

Reply:
There are some permissions that should be excluded from being renamed at this time. Currently, it's just these two: Reactions.Negative.Add and Reactions.Positive.Add. It's these two, in particular, because they don't really match the naming scheme of "resource.action". If we clip the addon namespace off, they become "negative.add" and "positive.add", which doesn't do much to convey what the permission actually does. We're also trying to avoid two-word sections (e.g. reactions.positiveAdd). These permission names will probably be revisited in the near future, but for the time being, they're edge cases that we shouldn't rename.

Reply:
What about all the other plugins permissions?

- Plugins.Debugger.View
- Plugins.Pockets.Manage
- Plugins.Flagging.Notify
- AvatarPool.CustomUpload.Allow
- Plugins.Attachments.Upload.Allow
- ....

Reply:
* Plugins.Debugger.View - becomes "debugger.view"
* Plugins.Pockets.Manage - becomes "pockets.manage"
* Plugins.Flagging.Notify - becomes "flagging.notify"
* AvatarPool.CustomUpload.Allow - "customupload.allow"
* Plugins.Attachments.Upload.Allow - Not renamed, because it's four parts.

Some of these aren't ideal, but they're much more indicative of what the permission actually is.